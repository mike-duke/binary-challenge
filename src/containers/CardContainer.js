import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Card } from '../components/Card';
import apiKey from '../apiKey';
import { fetchRelevantArticles } from '../thunks/fetchRelevantArticles';
import { fetchCurrentArticles } from '../thunks/fetchCurrentArticles';
import Nav from '../components/Nav';
import Filter from '../containers/Filter';
import uuid from 'uuid';

export class CardContainer extends Component {

  componentDidMount() {
    const relevantUrl = `https://newsapi.org/v2/everything?q=+(parent OR parents) AND +(kid OR kid OR child OR children) AND ${this.props.topic} &language=en&sortBy=relevancy&apiKey=${apiKey}&pageSize=100`;
    this.props.addRelevantArticlesToStore(relevantUrl);
    const currentUrl = 'https://newsapi.org/v2/top-headlines?country=us&apiKey=5b9c51c910284592a139f895c16d66ce';
    this.props.addCurrentArticlesToStore(currentUrl);
  }

  filterRelevantArticles = () => {
    return this.props.relevantArticles.filter((article) => {
      const { topic } = this.props;
      return article.description.includes(topic) || article.title.includes(topic) || article.content.includes(topic);
    }).map((article) => {
      return <Card key={uuid()} article={article} />
    });
  }
  
  render() {
    const { path } = this.props.match;
    let articlesToDisplay;
    switch(path) {
      case '/relevant':
        articlesToDisplay = this.filterRelevantArticles();
        break
      case '/current':
        articlesToDisplay = this.props.currentArticles.map((article) => {
          return <Card key={uuid()} article={article} />
        });
        break
      case '/saved': 
        articlesToDisplay = "saved articles";
        break
      default:
        break;  
    }

    if (this.props.isLoading) {
      return (
        <div className="card-container">
          <div className="loading">
            <img src="https://media.giphy.com/media/3oEjI6SIIHBdRxXI40/giphy.gif" alt="loading spinner"/>
          </div>
        </div>
      )
    } else {
      return (
        <section className="card-and-nav-container">
        <div className="card-container">
          {articlesToDisplay}
          <Nav />
        </div>
        <Filter path={path}/>
      </section>
      )
    }
  }
}

export const mapStateToProps = (state) => ({
  relevantArticles: state.relevantArticles,
  currentArticles: state.currentArticles,
  topic: state.topic,
  isLoading: state.isLoading
})

export const mapDispatchToProps = (dispatch) => ({
  addRelevantArticlesToStore: (url) => dispatch(fetchRelevantArticles(url)),
  addCurrentArticlesToStore: url => dispatch(fetchCurrentArticles(url))
});

export default connect(mapStateToProps, mapDispatchToProps)(CardContainer);

// if (isLoading) {
//   return (
//     <div className="loading">
//       <img src="https://media.giphy.com/media/3oEjI6SIIHBdRxXI40/giphy.gif" alt="loading spinner"/>
//     </div>
//   )
// }