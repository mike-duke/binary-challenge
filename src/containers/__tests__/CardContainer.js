import React from 'react';
import { shallow } from 'enzyme';
import { CardContainer } from '../CardContainer';

describe('CardContainer', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(
      <CardContainer 
        match={{path: '/relevant'}}
        relevantArticles={[]}
        addRelevantArticlesToStore={jest.fn()} 
        addCurrentArticlesToStore={jest.fn()} />);

    expect(wrapper).toMatchSnapshot();
  });
});