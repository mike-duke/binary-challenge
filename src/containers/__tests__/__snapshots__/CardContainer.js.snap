// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot if isLoading is true 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    addCurrentArticlesToStore={
      [MockFunction] {
        "calls": Array [
          Array [
            "https://newsapi.org/v2/top-headlines?country=us&apiKey=5b9c51c910284592a139f895c16d66ce",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    addDisplayedArticlesToStore={
      [MockFunction] {
        "calls": Array [
          Array [
            Array [],
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    addRelevantArticlesToStore={
      [MockFunction] {
        "calls": Array [
          Array [
            "https://newsapi.org/v2/everything?q=+(parent OR parents) AND +(kid OR kid OR child OR children) AND anxiety &language=en&sortBy=relevancy&apiKey=5b9c51c910284592a139f895c16d66ce&pageSize=100",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    currentArticles={Array []}
    isLoading={true}
    match={
      Object {
        "path": "/relevant",
      }
    }
    relevantArticles={Array []}
    topic="anxiety"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="loading"
      >
        <img
          alt="loading spinner"
          src="https://media.giphy.com/media/3oEjI6SIIHBdRxXI40/giphy.gif"
        />
      </div>,
      "className": "card-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="loading spinner"
          src="https://media.giphy.com/media/3oEjI6SIIHBdRxXI40/giphy.gif"
        />,
        "className": "loading",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "loading spinner",
          "src": "https://media.giphy.com/media/3oEjI6SIIHBdRxXI40/giphy.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="loading"
        >
          <img
            alt="loading spinner"
            src="https://media.giphy.com/media/3oEjI6SIIHBdRxXI40/giphy.gif"
          />
        </div>,
        "className": "card-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="loading spinner"
            src="https://media.giphy.com/media/3oEjI6SIIHBdRxXI40/giphy.gif"
          />,
          "className": "loading",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "loading spinner",
            "src": "https://media.giphy.com/media/3oEjI6SIIHBdRxXI40/giphy.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`CardContainer should match the snapshot when the path is "/current" 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    addCurrentArticlesToStore={
      [MockFunction] {
        "calls": Array [
          Array [
            "https://newsapi.org/v2/top-headlines?country=us&apiKey=5b9c51c910284592a139f895c16d66ce",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    addDisplayedArticlesToStore={[MockFunction]}
    addRelevantArticlesToStore={
      [MockFunction] {
        "calls": Array [
          Array [
            "https://newsapi.org/v2/everything?q=+(parent OR parents) AND +(kid OR kid OR child OR children) AND anxiety &language=en&sortBy=relevancy&apiKey=5b9c51c910284592a139f895c16d66ce&pageSize=100",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    currentArticles={
      Array [
        Object {
          "author": "simple simon",
          "content": "anxiety is a cat's companion",
          "description": "a greeting",
          "title": "hello world",
        },
      ]
    }
    isLoading={false}
    match={
      Object {
        "path": "/current",
      }
    }
    relevantArticles={Array []}
    topic="anxiety"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="container-and-nav"
        >
          <div
            className="container"
          >
            <Card
              article={
                Object {
                  "author": "simple simon",
                  "content": "anxiety is a cat's companion",
                  "description": "a greeting",
                  "title": "hello world",
                }
              }
            />
          </div>
          <Nav />
        </div>,
        <Connect(Filter)
          path="/current"
        />,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="container"
            >
              <Card
                article={
                  Object {
                    "author": "simple simon",
                    "content": "anxiety is a cat's companion",
                    "description": "a greeting",
                    "title": "hello world",
                  }
                }
              />
            </div>,
            <Nav />,
          ],
          "className": "container-and-nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Card
                  article={
                    Object {
                      "author": "simple simon",
                      "content": "anxiety is a cat's companion",
                      "description": "a greeting",
                      "title": "hello world",
                    }
                  }
                />,
              ],
              "className": "container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "acb25004-8363-46c9-99bd-3cb45a66eb21",
                "nodeType": "function",
                "props": Object {
                  "article": Object {
                    "author": "simple simon",
                    "content": "anxiety is a cat's companion",
                    "description": "a greeting",
                    "title": "hello world",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/current",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="container-and-nav"
          >
            <div
              className="container"
            >
              <Card
                article={
                  Object {
                    "author": "simple simon",
                    "content": "anxiety is a cat's companion",
                    "description": "a greeting",
                    "title": "hello world",
                  }
                }
              />
            </div>
            <Nav />
          </div>,
          <Connect(Filter)
            path="/current"
          />,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="container"
              >
                <Card
                  article={
                    Object {
                      "author": "simple simon",
                      "content": "anxiety is a cat's companion",
                      "description": "a greeting",
                      "title": "hello world",
                    }
                  }
                />
              </div>,
              <Nav />,
            ],
            "className": "container-and-nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Card
                    article={
                      Object {
                        "author": "simple simon",
                        "content": "anxiety is a cat's companion",
                        "description": "a greeting",
                        "title": "hello world",
                      }
                    }
                  />,
                ],
                "className": "container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "acb25004-8363-46c9-99bd-3cb45a66eb21",
                  "nodeType": "function",
                  "props": Object {
                    "article": Object {
                      "author": "simple simon",
                      "content": "anxiety is a cat's companion",
                      "description": "a greeting",
                      "title": "hello world",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/current",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`CardContainer should match the snapshot when the path is "/relevant" 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    addCurrentArticlesToStore={
      [MockFunction] {
        "calls": Array [
          Array [
            "https://newsapi.org/v2/top-headlines?country=us&apiKey=5b9c51c910284592a139f895c16d66ce",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    addDisplayedArticlesToStore={
      [MockFunction] {
        "calls": Array [
          Array [
            Array [
              Object {
                "author": "simple simon",
                "content": "anxiety is a cat's companion",
                "description": "a greeting",
                "title": "hello world",
              },
            ],
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    addRelevantArticlesToStore={
      [MockFunction] {
        "calls": Array [
          Array [
            "https://newsapi.org/v2/everything?q=+(parent OR parents) AND +(kid OR kid OR child OR children) AND anxiety &language=en&sortBy=relevancy&apiKey=5b9c51c910284592a139f895c16d66ce&pageSize=100",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    currentArticles={Array []}
    isLoading={false}
    match={
      Object {
        "path": "/relevant",
      }
    }
    relevantArticles={
      Array [
        Object {
          "author": "simple simon",
          "content": "anxiety is a cat's companion",
          "description": "a greeting",
          "title": "hello world",
        },
      ]
    }
    topic="anxiety"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="container-and-nav"
        >
          <div
            className="container"
          >
            <Card
              article={
                Object {
                  "author": "simple simon",
                  "content": "anxiety is a cat's companion",
                  "description": "a greeting",
                  "title": "hello world",
                }
              }
            />
          </div>
          <Nav />
        </div>,
        <Connect(Filter)
          path="/relevant"
        />,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="container"
            >
              <Card
                article={
                  Object {
                    "author": "simple simon",
                    "content": "anxiety is a cat's companion",
                    "description": "a greeting",
                    "title": "hello world",
                  }
                }
              />
            </div>,
            <Nav />,
          ],
          "className": "container-and-nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Card
                  article={
                    Object {
                      "author": "simple simon",
                      "content": "anxiety is a cat's companion",
                      "description": "a greeting",
                      "title": "hello world",
                    }
                  }
                />,
              ],
              "className": "container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "44fe6463-9a69-4c9d-bf5e-68d878ef4943",
                "nodeType": "function",
                "props": Object {
                  "article": Object {
                    "author": "simple simon",
                    "content": "anxiety is a cat's companion",
                    "description": "a greeting",
                    "title": "hello world",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/relevant",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="container-and-nav"
          >
            <div
              className="container"
            >
              <Card
                article={
                  Object {
                    "author": "simple simon",
                    "content": "anxiety is a cat's companion",
                    "description": "a greeting",
                    "title": "hello world",
                  }
                }
              />
            </div>
            <Nav />
          </div>,
          <Connect(Filter)
            path="/relevant"
          />,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="container"
              >
                <Card
                  article={
                    Object {
                      "author": "simple simon",
                      "content": "anxiety is a cat's companion",
                      "description": "a greeting",
                      "title": "hello world",
                    }
                  }
                />
              </div>,
              <Nav />,
            ],
            "className": "container-and-nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Card
                    article={
                      Object {
                        "author": "simple simon",
                        "content": "anxiety is a cat's companion",
                        "description": "a greeting",
                        "title": "hello world",
                      }
                    }
                  />,
                ],
                "className": "container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "44fe6463-9a69-4c9d-bf5e-68d878ef4943",
                  "nodeType": "function",
                  "props": Object {
                    "article": Object {
                      "author": "simple simon",
                      "content": "anxiety is a cat's companion",
                      "description": "a greeting",
                      "title": "hello world",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/relevant",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`CardContainer should match the snapshot when there is no path 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    addCurrentArticlesToStore={
      [MockFunction] {
        "calls": Array [
          Array [
            "https://newsapi.org/v2/top-headlines?country=us&apiKey=5b9c51c910284592a139f895c16d66ce",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    addDisplayedArticlesToStore={[MockFunction]}
    addRelevantArticlesToStore={
      [MockFunction] {
        "calls": Array [
          Array [
            "https://newsapi.org/v2/everything?q=+(parent OR parents) AND +(kid OR kid OR child OR children) AND anxiety &language=en&sortBy=relevancy&apiKey=5b9c51c910284592a139f895c16d66ce&pageSize=100",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    currentArticles={Array []}
    isLoading={false}
    match={
      Object {
        "path": "",
      }
    }
    relevantArticles={Array []}
    topic="anxiety"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="container-and-nav"
        >
          <div
            className="container"
          />
          <Nav />
        </div>,
        <Connect(Filter)
          path=""
        />,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="container"
            />,
            <Nav />,
          ],
          "className": "container-and-nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "container",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="container-and-nav"
          >
            <div
              className="container"
            />
            <Nav />
          </div>,
          <Connect(Filter)
            path=""
          />,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="container"
              />,
              <Nav />,
            ],
            "className": "container-and-nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "container",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
